ciao-controller/openstack_volume.go

```diff
@@ -233,7 +233,36 @@ func (c *controller) ListVolumes(tenant string) ([]block.ListVolume, error) {
 }
 
 func (c *controller) ListVolumesDetail(tenant string) ([]block.VolumeDetail, error) {
-	return make([]block.VolumeDetail, 0), nil
+	var vols []block.VolumeDetail
+
+	devs, err := c.ds.GetBlockDevices(tenant)
+	if err != nil {
+		return vols, err
+	}
+
+	for _, data := range devs {
+		var vol block.VolumeDetail
+
+		vol.ID = data.ID
+		vol.Size = data.Size
+		vol.OSVolTenantAttr = data.TenantID
```

Comment:
this one we don't have to check the tenant ID on because the call to GetBlockDevices() uses the tenant as an argument. You do need to check on ListVolumeDetail() because the argument is simply the volume ID and the call to get the block device by volume ID doesn't look at the tenant.